src/python/pants/base/revision.py

```diff
@@ -87,21 +91,24 @@ def components(self):
     """
     return list(self._components)
 
-  def __cmp__(self, other):
-    for ours, theirs in izip_longest(self._components, other._components, fillvalue=0):
-      difference = cmp(ours, theirs)
-      if difference != 0:
-        return difference
-    return 0
+  def _is_valid_operand(self, other):
+    return hasattr(other, '_components')
 
   def __repr__(self):
     return '{}({})'.format(self.__class__.__name__, ', '.join(map(repr, self._components)))
 
   def __eq__(self, other):
-    return hasattr(other, '_components') and tuple(self._components) == tuple(other._components)
-
-  def __ne__(self, other):
-    return not self.__eq__(other)
+    if not self._is_valid_operand(other):
```

Comment:
This slightly changes the behavior from before. When comparing to non-Revisions, this returns `NotImplemented`, which is the expected behavior. instead of the prior `False`.

Reply:
Is there any python3 compat reason to mix in this behavior change? As an example of future conversion PRs, it would be good to stick to pure semantic-preserving automated conversion if possible.

Reply:
I'd prefer it if just to keep semantics changes out of the conversion PRs. Especially since `Revision` is [`:API: public`](https://github.com/Eric-Arellano/pants/blob/06b0db028a0da65eb1d474a4f2dcc393e6aaa283/src/python/pants/base/revision.py#L20) and the semantic shift is `False` -> raise. We can't know how this will break API users.

Reply:
More on `:API: public` here: https://www.pantsbuild.org/deprecation_policy.html

Reply:
I reverted `__eq__` to return `False`

But am keeping `__lt__` as returning `NotImplemented`. The prior implementation in `__cmp__` would return `AttributeError` if `other._components` is defined.

Reply:
> ... and the semantic shift is False -> raise.

I was mistaken there. The semantic shift is subtle to non-existant depending on how `other` implements `__eq__`. That said, still a fan of nixing improvements in these PRs. TODO is ok, but it really should be attributed and / or - better - assigned an issue with the issue link noted in the TODO. Your spirit of wanting to fix is admirable and I'm sure you'll see more of these. Perhaps just start 1 umbrella issue to capture all the small TODO's you find then you or someone else can circle back and do the fixes in unrelated PR's.

Reply:
Okay I reverted everything back to identical semantics from before (just more explicit). 

I think it's a good policy to avoid any changes in these PRs because they get buried. Instead, will add notes in form `TODO(python3port)` and open an issue at https://github.com/pantsbuild/pants/issues/6071.